compiler/rustc_mir_transform/src/add_retag.rs

```diff
@@ -88,10 +79,8 @@ impl<'tcx> MirPass<'tcx> for AddRetag {
         };
         let place_base_raw = |place: &Place<'tcx>| {
             // If this is a `Deref`, get the type of what we are deref'ing.
-            let deref_base =
-                place.projection.iter().rposition(|p| matches!(p, ProjectionElem::Deref));
-            if let Some(deref_base) = deref_base {
-                let base_proj = &place.projection[..deref_base];
+            if place.ret_deref().is_some() {
+                let base_proj = &place.projection[..0];
```

Comment:
this just gives us an empty slice

Reply:
This means you can just ask for the type of the `place.local`, no need to use `Place::ty_from`